client/components/forms/form-text-input/test/index.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import chai, { expect } from 'chai';
+import { shallow } from 'enzyme';
+import { spy } from 'sinon';
+import chaienzyme from 'chai-enzyme';
+import FormTextInput from 'components/forms/form-text-input';
+
+chai.use( chaienzyme() );
+
+describe( '<FormTextInput />', () => {
+	describe( 'rendering', () => {
+		it( 'should add the provided class names', () => {
+			const wrapper = shallow( <FormTextInput className="test" isError isValid /> );
+
+			expect( wrapper ).to.have.className( 'test' );
+			expect( wrapper ).to.have.className( 'is-error' );
+			expect( wrapper ).to.have.className( 'is-valid' );
+		} );
+
+		it( 'should not pass props down to the form fields', () => {
+			const wrapper = shallow( <FormTextInput isValid isError selectOnFocus /> );
+
+			expect( wrapper ).to.not.have.prop( 'isValid' );
+			expect( wrapper ).to.not.have.prop( 'isError' );
+			expect( wrapper ).to.not.have.prop( 'selectOnFocus' );
+		} );
+
+		it( 'should call select if selectOnFocus is true', () => {
+			const wrapper = shallow( <FormTextInput selectOnFocus={ true } /> );
+			const event = {
+				target: {
+					select: () => {}
+				}
+			};
+
+			spy( event.target, 'select' );
+			wrapper.simulate( 'click', event );
+
+			expect( event.target.select ).to.have.been.calledOnce;
+		} );
+
+		it( 'should not call select if selectOnFocus is false', () => {
+			const wrapper = shallow( <FormTextInput selectOnFocus={ false } /> );
+			const event = {
+				target: {
+					select: () => {}
+				}
+			};
+
+			spy( event.target, 'select' );
+			wrapper.simulate( 'click', event );
+
+			expect( event.target.select ).to.not.have.been.called;
+		} );
+
```

Comment:
```
wp-calypso (add/form-text-input-tests) $ ./node_modules/.bin/eslint client/components/forms/form-text-input/test/index.jsx 

/Users/andrew/Documents/Code/wp-calypso/client/components/forms/form-text-input/test/index.jsx
  13:44  warning  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
  57:2   warning  Block must not be padded by blank lines                                    padded-blocks
  59:1   warning  Block must not be padded by blank lines                                    padded-blocks

✖ 3 problems (0 errors, 3 warnings)
```

The `className` warning is something of a false positive. I'll see if I can improve this rule to ignore test files.